LiskHQ/lisk-hub
# Convert Lisk Hub to Progressive Web Application - Close #509

### What was the problem?
Lisk Hub is not a PWA
### How did I fix it?
Add and configured next packages:
[webpack-pwa-manifest](https://github.com/arthurbergmz/webpack-pwa-manifest) - to generate a manifest file
[offline-plugin](https://github.com/NekR/offline-plugin) - to configure service worker and app cache 

### How to test it?
1) Build project with `production` flag.
2) Run server with Lisk Hub
3) Chrome on a computer:  `Dev tools` -> `Application` -> `Manifest` -> `Add to homescreen`
4) Turn off internet connection
5) Refresh page

To check cache update:
1) Change source file (add `alert(1)` for example)
2) Build new version
3) Reload the webpage
4) Close the tab of the webpage (make sure there are no more open tabs of this website)
5) Open that webpage again in a new tab
6) Check `alert(1)`

I haven't any `Android` device to check how it will work on it, but this is what I got on `iPhone` with the latest ios `11.3`:

<img src="https://user-images.githubusercontent.com/922338/38274926-ee55e0b2-3798-11e8-86ad-8861f74fc1d8.png" width="300px" align="left"/>
<img src="https://user-images.githubusercontent.com/922338/38274932-f44759d8-3798-11e8-94c1-cce1df20a1cf.png" width="300px" />

### Todo?
- [ ] Add logo with high resolution and convert to small sizes.
- [ ] Add message when a device is `offline`?
- [ ] Add message when the new version is available?
- [ ] Add splash into the manifest file for mobile devices?

Any suggestion?
### Review checklist
- The PR solves #509
- All new code follows best practices
